perfzero/lib/cloud_manager.py

```diff
@@ -217,171 +248,146 @@ def list_all(project):
-        types, run 'gcloud compute machine-types list --project=${project_name}'
-        ''')
-    parser.add_argument(
-        '--image',
-        default='tf-ubuntu-1604-20180927-410',
-        type=str,
-        help='''Specifies the name of the image that the disk will be initialized with.
-        A new disk will be created based on the given image. To view a list of
-        public images and projects, run 'gcloud compute images list --project=${project_name}'. It is best
-        practice to use image when a specific version of an image is needed.
-        ''')
-    parser.add_argument(
-        '--nvme_count',
-        default=0,
-        type=int,
-        help='''Specifies the number of NVME local SSD devices to attach to the instance.
-        '''
-        )
-
-  flags, unparsed = parser.parse_known_args(argv)  # pylint: disable=redefined-outer-name
-  if unparsed:
-    logging.error('Arguments %s are not recognized', unparsed)
+  flags = parser.parse_args()
+
+  if not hasattr(sys.modules[__name__], flags.command):
+    print('Error: The command <{}> is not recognized\n'.format(flags.command))
+    parser.print_help()
     sys.exit(1)
 
+  if flags.command == 'create':
+    if not flags.machine_type or not flags.image:
+      raise ValueError('Need to specify machine_type and image for creating '
+                       'instances')
+
   level = logging.DEBUG if flags.debug else logging.INFO
   logging.basicConfig(format='%(message)s', level=level)
 
   return flags
-if __name__ == '__main__':
-  parser = argparse.ArgumentParser(
-      usage='''cloud_manager.py <command> [<args>]
 
-The supported commands are:
-  create:  Create a computing instance in gcloud that is unique to the specified username, which is your ldap by default
-  start:   Start the computing instance in gcloud that is unique to the specified username, which is your ldap by default
-  stop:    Stop the computing instance in gcloud that is unique to the specified username, which is your ldap by default
-  delete:  Delete the computing instance in gcloud that is unique to the specified username, which is your ldap by default
-  status:  Query the status and information of the computing instance in gcloud that is unique to the specified username, which is your ldap by default
-  list_all:    Query the status of all computing instances that are created by this script.'''
-  )
-  parser.add_argument(
-      'command',
-      type=str
-      )
 
-  flags = parser.parse_args(sys.argv[1:2])
+if __name__ == '__main__':
+  flags = parse_arguments()
   command = flags.command
-  if not hasattr(sys.modules[__name__], command):
-    print('Error: The command <{}> is not recognized\n'.format(command))
-    parser.print_help()
-    sys.exit(1)
-
-  flags = parse_arguments(sys.argv[2:], command)
-
   if command == 'create':
-    create(flags.username, flags.project, flags.zone, flags.machine_type,
+    create(flags.instance_name, flags.project, flags.zone, flags.machine_type,
```

Comment:
The benefit of using `user_name` is that we can easily identify who created the instance. So that if some instances are running but not used for a long term, we can find the owner and possibly stop those instances. Using `user_name` as parameter name will make this goal clearer.

If there is no specific benefit of letting user user arbitrary instance name string, may be we can keep it as is.